File: pattern/12.cpp
// 1                1
// 12            21
// 123        321
// 1234    4321
// 1234554321

#include <iostream>
using namespace std;

int main() {
    int rows = 4;

    for (int i = 1; i <= rows; i++) {
        
        for (int j = 1; j <= i; j++) {
            cout << j;
        }
             for (int j = 1; j <= 2 * (rows - i); j++) {
            cout << " ";
        }

      
        for (int j = i; j >= 1; j--) {
            cout << j;
        }

        
     cout << endl;
        }
        return 0;
}

File: pattern/12.py
# Number of rows
# since rows = 4

for i in range(1, 5):
    # Print numbers from 1 to i
    for j in range(1, i + 1):
        print(j, end="")

    # Print spaces in between
    print(" " * (2 * (4 - i)), end="")

    # Print numbers from i to 1
    for j in range(i, 0, -1):
        print(j, end="")

    # Move to the next row
    print()
